File: src/integrationTest/java/org/revcloud/integration/KickTest.java
package org.revcloud.integration;

import java.util.Map;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.revcloud.input.Kick;

class KickTest {
  @Test
  @DisplayName("Test Default values")
  void defaultValues() {
    final var kickOffConfig =
        Kick.configure()
            .templatePath("templatePath")
            .environmentPath(null)
            .itemNameToSuccessType(Map.of())
            .off();
    System.out.println(kickOffConfig);
  }
}

File: src/test/kotlin/com/salesforce/revoman/input/KickTest.kt
package com.salesforce.revoman.input

import io.kotest.assertions.throwables.shouldThrow
import org.junit.jupiter.api.Test

class KickTest {
  @Test
  fun `'runOnlySteps' and 'skipSteps' cannot have intersection`() {
    val exception =
      shouldThrow<IllegalArgumentException> {
        Kick.configure().skipSteps("a", "b").runOnlySteps("b", "c").off()
      }
  }

  @Test
  fun `'haltOnAnyFailureExceptForSteps' should be empty when 'haltOnAnyFailure' is set to True`() {
    val exception =
      shouldThrow<IllegalArgumentException> {
        Kick.configure().haltOnAnyFailure(true).haltOnAnyFailureExceptForSteps("a", "b").off()
      }
  }
}
